types/src/on_chain_config.rs

```diff
@@ -78,6 +78,15 @@ pub trait OnChainConfig: Send + Sync + DeserializeOwned {
     fn deserialize_into_config(bytes: &[u8]) -> Result<Self> {
         Self::deserialize_default_impl(bytes)
     }
+
+    fn fetch_config<T>(storage: T) -> Option<Self>
+    where
+        T: ConfigStorage,
+    {
+        storage
+            .fetch_config(Self::CONFIG_ID.access_path())
+            .and_then(|bytes| Self::deserialize_into_config(&bytes).ok())
```

Comment:
Sorry I removed the `error!` log in my previous PR. Should we add it back here?

Reply:
we can add logging at the function callsite for better context (vs inside this function) - can do it in this PR